### PR TITLE
Add samplot 1.0.1

### PR DESCRIPTION
**Pull request by cbrueffer**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/samplot/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/ryanlayer/samplot/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256hash }}
+
+build:
+  number: 0
+  skip: True  # [osx]
```
Comment (npavlovikj) [inline recipes/samplot/meta.yaml]: Is there a reason why OSX is skipped? All Python packages are available for OSX as well.

File: recipes/samplot/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
[the remark this answers is not in the document]
Comment (cbrueffer) [reply recipes/samplot/meta.yaml]: A remnant of the recipe I used as template, I'll remove it.

File: recipes/samplot/meta.yaml (hunk 1 above)
Comment (cbrueffer) [reply recipes/samplot/meta.yaml]: Same as above, a remnant that can be removed. Thanks for pointing those out!

**Comment by cbrueffer** [comment]
Using `noarch: generic` here, since the code does not follow Python package conventions (source basically consists of a src folder with the two scripts). Is it still possible/advisable to use `noarch: python`?

**Comment by dpryan79** [comment]
`noarch: python` is only relevant for actual python packages (it triggers post-install scripts for moving the lib directory to `lib/python3.6/site-packages`, `lib/python3.7/site-packages`, etc.), so don't worry about it.

**Comment by cbrueffer** [comment]
Great, thanks!

**Comment by npavlovikj** [comment]
Thanks @cbrueffer and @dpryan79  :)

**Pull request merged**